### PR TITLE
Fix `doRangeFormat` to work with all kind of user text selection

### PR DESCRIPTION
**Pull request by razronen**

### CODE REVIEW

File: src/languageService/kustoLanguageService.ts [hunk 1]
```diff
@@ -426,22 +426,28 @@ class KustoLanguageService implements LanguageService {
...
-        );
+        const rangeStartOffset: number = document.offsetAt(range.start);
+        const rangeEndOffset: number = document.offsetAt(range.end);
+        const commands = this.getCommandsInDocument(document).then(commands => {
+            const commandsInRange = commands.map(command => {
+                // Chose only selected text from command.
+                const commandSelectionStart = Math.max(rangeStartOffset, command.absoluteStart) - command.absoluteStart;
+                const commandSelectionEnd = Math.min(rangeEndOffset, command.absoluteEnd) - command.absoluteStart;
+
+                command.text = commandSelectionStart > commandSelectionEnd ? "" : command.text.substring(commandSelectionStart, commandSelectionEnd);
```
[the remark this answers is not in the document]
Comment (razronen) [reply src/languageService/kustoLanguageService.ts:437]: This is a very corner case I think might happen when the prefix of a query is a bunch of spaces and the user select some of those spaces deliberately. I didn’t test it though but just wanted to make sure this weird circumstances doesn’t result in something even more strange I saw:
If .substring() method second param is smaller then the first one it automatically switches them
